workspace/solution.go

```diff
@@ -87,3 +95,22 @@ func (s *Solution) PathToParent() string {
 	}
 	return filepath.Join(dir, s.Track)
 }
+
+func createIgnoreSubdir(path string) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		if err := os.Mkdir(path, os.FileMode(0755)); err != nil {
+			return fmt.Errorf("failed to create directory: %s", path)
+		}
+	}
+	return nil
+}
+
+func migrateLegacySolutionFile(path string, legacySolutionPath string, solutionPath string) error {
+	if err := createIgnoreSubdir(filepath.Join(path, IgnoreSubdir)); err != nil {
+		return err
+	}
+	if err := os.Rename(legacySolutionPath, solutionPath); err != nil {
+		return errors.New("failed migrating legacy solution file")
```

Comment:
This doesn't actually end up getting handled (it gets called in the path traversal loop in `workspace.SolutionDir`). Would an acceptable solution be to `panic`?